routes/api/url.js

```diff
@@ -27,15 +27,25 @@ router.get("/", middleAuth, async (req, res) => {
 });
 
 router.post("/compress", middleAuth, async (req, res) => {
-  const { longUrl, lastDate } = req.body;
+  const { longUrl, lastDate, customCode } = req.body;
 
-  // const baseUrl = "http://localhost:5000";
+  // base url = "http://localhost:5000"
   const baseUrl = process.env.BASE_URL;
   if (!validUrl.isUri(baseUrl)) {
     return res.status(400).send({ msg: "Invalid Base Url" });
   }
 
-  const urlCode = shortId.generate();
+  let urlCode;
+  if (customCode) {

```

Comment:
Hey @joaovitorzv 
Add this custom code checking after valid URL check and checking if the URL already exists. Since there is no meaning of checking if the Long URL is not valid or the URL already exists in the DB.